dns/resolver.go

```diff
@@ -268,6 +268,11 @@ func NewUDPResolver(pd transport.PacketDialer, resolverAddr string) Resolver {
 			return nil, &nestedError{ErrDial, err}
 		}
 		defer conn.Close()
+		// force close connection is context is done/cancelled.
```

Comment:
@fortuna I have to add this here to close the connection for pending receives (in case of timeouts) when another attempt succeeds. I managed to do this with the following change. I am basically cancelling the passed context in `connectivity.go` and checking if context is done. This is the best I came up with but I am not sure if there's a more elegant way to accomplish this.